src/backend/base/langflow/services/database/models/api_key/crud.py

```
@@ -50,17 +49,14 @@
-
-
 async def check_key(session: AsyncSession, api_key: str) -> User | None:
     """Check if the API key is valid."""
     query: SelectOfScalar = select(ApiKey).options(selectinload(ApiKey.user)).where(ApiKey.api_key == api_key)
     api_key_object: ApiKey | None = (await session.exec(query)).first()
     if api_key_object is not None:
-        task = asyncio.create_task(update_total_uses(api_key_object.id))
-        task.add_done_callback(update_total_uses_tasks.discard)
-        update_total_uses_tasks.add(task)
+        settings_service = get_settings_service()
+        if settings_service.settings.disable_track_apikey_usage != True:
```

Ruff (0.11.9)

Avoid inequality comparisons to `True`; use `if not ...:` for false checks

Replace comparison

(E712)